### PR TITLE
[RG-56] Implement recent projects action

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Implement recent projects feature

### Describe the technical details
Settings file is created or opened when app start. All recent projects are saved to setting file. 10 projects maximum 
and 5 projects on welcome page

 ### Which part of the code base require a change
 
 - [ ] Library: <Specify the library name>
 - [ ] Plug-in: <Specify the plugin name>
 - [x] Engine
 - [ ] Documentation
 - [ ] Regression tests
 - [ ] Continous Integration (CI) scripts

### Screenshots

image
image

### CODE REVIEW

**Review by TarasSydor — APPROVED** [review]
looks good.

**Review by ravic-rs — COMMENTED** [review]
Pending technical review with team.

File: src/MainWindow/main_window.cpp [hunk 1]
```diff
@@ -228,6 +227,39 @@ void MainWindow::cleanUpDockWidgets(std::vector<QDockWidget*>& dockWidgets) {
...
+  ReShowWindow(project);
+  loadFile(project);
+}
+
+void MainWindow::saveToRecentSettings(const QString& project) {
+  if (project.isEmpty()) return;
+
+  std::list<QString> projects;
+
+  for (uint i = 0; i < 10; i++) {
```
Comment (ravic-rs) [inline src/MainWindow/main_window.cpp:240]: Consider moving the number of recent projects (10) and (5) outside this function which can be set explicitly.

**Review by ravic-rs — COMMENTED** [review]
One comment to address.

**Comment by ravic-rs** [comment]
@TarasSydor FYI, this is going to be merged today.

File: src/MainWindow/main_window.cpp (hunk 1 above)
Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp]: done

File: src/MainWindow/main_window.cpp [hunk 2]
```diff
@@ -284,7 +319,28 @@ void MainWindow::createProgressBar() {
   m_progressWidget->hide();
 }
 
+void MainWindow::createRecentMenu() {
+  for (auto it{m_recentProjectsActions.begin()};
+       it != m_recentProjectsActions.end(); ++it)
+    delete it->first;
+  m_recentProjectsActions.clear();
+
+  for (int projCounter = 9; projCounter >= 0; projCounter--) {
```
Comment (ravic-rs) [inline src/MainWindow/main_window.cpp:328]: This part seems to be left out from re-factoring.

Comment (KochynVolodymyr) [reply src/MainWindow/main_window.cpp]: Done

**Review by ravic-rs — APPROVED** [review]
Looks good, waiting for CI tests.

**Pull request merged**